conformance/packages/dns-test/src/docker/hickory.Dockerfile

```diff
@@ -1,15 +1,18 @@
 FROM rust:1-slim-bookworm
 
 # ldns-utils = ldns-{key2ds,keygen,signzone}
+# pkg-config + libssl-dev are needed to build hickory-dns with +dnssec-openssl
 RUN apt-get update && \
     apt-get install -y \
+        pkg-config \
+        libssl-dev \
         ldnsutils \
         tshark
 
 # `dns-test` will invoke `docker build` from a temporary directory that contains
 # a clone of the hickory repository. `./src` here refers to that clone; not to
 # any directory inside the `dns-test` repository
 COPY ./src /usr/src/hickory
-RUN cargo install --path /usr/src/hickory/bin --features recursor,dnssec-ring --debug && \
+RUN cargo install --path /usr/src/hickory/bin --features recursor,dnssec-openssl --debug && \
```

Comment:
alternatively, we can continue using the `dnssec-ring` feature and instead switch the algorithm used to sign zone files in conformance tests from RSASHA1_NSEC3 to RSASHA256. any preference?

currently the algorithm is [hardcoded in the `dns-test` library](https://github.com/hickory-dns/hickory-dns/blob/02a8668fd54be5c3b1a9f8d8e79393229ffea2be/conformance/packages/dns-test/src/name_server.rs#L221) but we'll probably want to make that configurable on a test basis. that would let us add a regression test for #2252 if we do switch the algorithm to RSASHA256

Reply:
We hsould probably switch to SHA256 anyway since SHA-1 is deprecated. Would definitely prefer to continue relying on ring (and maybe aws-lc-rs later?) over OpenSSL.